platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java

```diff
@@ -610,37 +609,41 @@ private Collection<SwirldsPlatform> createLocalPlatforms(
             @NonNull final ApplicationDefinition appDefinition,
             @NonNull final Map<NodeId, Crypto> crypto,
             @NonNull final InfoSwirld infoSwirld,
-            @NonNull final Map<Long, SwirldMain> appMains,
+            @NonNull final Map<NodeId, SwirldMain> appMains,
             @NonNull final Configuration configuration,
             @NonNull final MetricsProvider metricsProvider) {
-        throwArgNull(appDefinition, "the app definition must not be null");
-        throwArgNull(crypto, "the crypto array must not be null");
-        throwArgNull(infoSwirld, "the infoSwirld must not be null");
-        throwArgNull(appMains, "the appMains map must not be null");
-        throwArgNull(configuration, "the configuration must not be null");
-        throwArgNull(metricsProvider, "the metricsProvider must not be null");
+        Objects.requireNonNull(appDefinition, "the app definition must not be null");
+        Objects.requireNonNull(crypto, "the crypto array must not be null");
+        Objects.requireNonNull(infoSwirld, "the infoSwirld must not be null");
+        Objects.requireNonNull(appMains, "the appMains map must not be null");
+        Objects.requireNonNull(configuration, "the configuration must not be null");
+        Objects.requireNonNull(metricsProvider, "the metricsProvider must not be null");
 
         final List<SwirldsPlatform> platforms = new ArrayList<>();
 
         final AddressBook addressBook = appDefinition.getAddressBook();
 
         int ownHostIndex = 0;
 
-        for (int i = 0; i < addressBook.getSize(); i++) {
-            final Address address = addressBook.getAddress(i);
+        for (final Address address : addressBook) {
+            final NodeId nodeId = address.getNodeId();
+            final int instanceNumber = addressBook.getIndexOfNodeId(nodeId);
+            if (!appMains.containsKey(nodeId)) {
```

Comment:
Why is this check needed, if we're also checking `if (address.isOwnHost())` directly below?

Reply:
You're right the calculation of the `appMains` includes a check of `address.isOwnHost()`.  I can remove the later check.